pkg/gameserverallocations/allocation_cache.go

```diff
@@ -67,9 +67,8 @@ func NewAllocationCache(informer informerv1.GameServerInformer, counter *gameser
 	}
 
 	// if we can do state filtering, then cache both Ready and Allocated GameServers
-	if runtime.FeatureEnabled(runtime.FeatureStateAllocationFilter) {
 		c.matcher = readyOrAllocatedGameServerMatcher
-	}
+
```

Comment:
We could clean up up more, and replace the matcher above on line 66 directly with the readyOrAllocatedGameServerMatcher -- and we could delete the `readyGameServerMatcher` function entirely.

(There's possibly a good reason to get rid of `matcher` as a data structure entirely, since we don't need to switch it out at runtime anymore).

Reply:
> 

Yeah, I forgot it. Thanks for reminding i will correct that.